WalletWasabi/WabiSabi/Client/CoinJoinCoinSelector.cs

```diff
@@ -47,9 +47,11 @@ public class CoinJoinCoinSelector
 			SecureRandom.Instance);
 
 	/// <param name="liquidityClue">Weakly prefer not to select inputs over this.</param>
-	public ImmutableList<TCoin> SelectCoinsForRound<TCoin>(IEnumerable<TCoin> coins, UtxoSelectionParameters parameters, Money liquidityClue)
+	public ImmutableList<TCoin> SelectCoinsForRound<TCoin>(IEnumerable<TCoin> coins, UtxoSelectionParameters parameters, Money liquidityClue, WasabiRandom? selectorRnd = null)
```

Comment:
Does it make sense to have two randoms in this function?

Reply:
I think it doesn't. `CoinJoinCoinSelector.SelectCoinsForRound()` should always use `CoinJoinCoinSelector.Rnd`.

Reply:
I would mock the random generator that is passed in the constructor of `CoinJoinCoinSelector` (for example on [this](https://github.com/zkSNACKs/WalletWasabi/pull/10879/files#diff-eabe5b799e533eccf51cd0edead0a39e0f08510b92799dbb44cba0f7ea4a6f06R21) line).

Reply:
I did, see https://github.com/trezor/WalletWasabi/tree/onvej-sl/get-rid-of-selectorrnd.

Reply:
By the way, the implementation of `ConfigureRng` doesn't make sense, because `ConfigureRng(returnValue: 1).GetInt(fromInclusive: 0, toExclusive: 1)` returns `1` which is not below `toExclusive`. This can break for example `LinqExtensions.RandomElement()`.

Reply:
Generally, my opinion is that a test that tests implementation rather than behaviour is not a good test because you have to check the test makes sense whenever you change the implementation (even if the behaviour remains the same).

Regarding the test you mentioned I think
  * the assertion in the test holds true for any value of the input randomness and
  * the sensitivity of the test is roughly equal for all values of the input randomness.
 
 This makes me believe that what I proposed should be sufficient for this particular test.

Both of you proposals make sense to me. I'm adding two more proposals:
  * Subclass `WasabiRandom` and extract `GetInputTarget` (or other parts of `GetInputTarget`) to this new class.
  * Run the tests in `CoinJoinCoinSelectionTests.cs` repeatedly with different values of the input randomness. I like this concept because it avoids difficult maintenance of very targeted tests for functions that frequently changes.